Evaluation/apply_training.py

```diff
@@ -0,0 +1,81 @@
+import os
+import sys
+import json
+from tqdm import tqdm
+
+import uproot
+import numpy as np
+import pandas as pd
+from tensorflow.keras.models import load_model
+
+import mlflow
+import hydra
+from hydra.utils import to_absolute_path
+from omegaconf import DictConfig, OmegaConf
+
+sys.path.insert(0, "../Training/python")
+import DataLoader
+
+@hydra.main(config_path='.', config_name='apply_training')
+def main(cfg: DictConfig) -> None:
+    # set up paths
+    mlflow.set_tracking_uri(f"file://{to_absolute_path(cfg.path_to_mlflow)}")
+    path_to_artifacts = to_absolute_path(f'{cfg.path_to_mlflow}/{cfg.experiment_id}/{cfg.run_id}/artifacts/')
+
+    # load the model
+    with open(to_absolute_path(f'{path_to_artifacts}/input_cfg/metric_names.json')) as f:
+        metric_names = json.load(f)
+    path_to_model = f'{path_to_artifacts}/model'
+    model = load_model(path_to_model, {name: lambda _: None for name in metric_names.keys()})
+
+    # load baseline training cfg and update it with parsed arguments
+    training_cfg = OmegaConf.load(to_absolute_path(cfg.path_to_training_cfg))
+    if cfg.training_cfg_upd is not None:
+        training_cfg = OmegaConf.merge(training_cfg, cfg.training_cfg_upd)
+    training_cfg = OmegaConf.to_object(training_cfg)
+
+    # instantiate DataLoader and get generator
+    scaling_cfg  = to_absolute_path(cfg.scaling_cfg)
+    dataloader = DataLoader.DataLoader(training_cfg, scaling_cfg)
+    gen_predict = dataloader.get_predict_generator()
+    tau_types_names = training_cfg['Setup']['tau_types_names']
```

Comment:
I think the problem that you described before related to the fact that `${path_to_mlflow}/${experiment_id}/${run_id}/artifacts/input_cfg/training_cfg.yaml` requires few more variables, in this case `self.dataloader_core = config["Setup"]["dataloader_core"]` that is not present in the initial training_v1.yaml. I think it can be solved by simple adding of the variable in the training artifacts/input_cfg. On the other hand, we will always modify data loader, and the freshest version of the Evaluation should use the freshest version of the data loader, for this we can 1) take into account it when modifying data loader 2) you can just merge the configs with the priority of config in artifacts. That would  add missing variables but also avoid changes in setups.

Reply:
Exactly, adding more variables/files to artifact cfg files isn't a maintable approach as we evolve with DataLoader development. So what I think should work is within the eval script do `git checkout <commit-hash>` where `commit-hash` is taken from artifacts and it corresponds to the commit used to run the training. This seems to me like a more general approach to synchronise training and evaluation steps: during training we log training/runtime specific files (fetched as artifacts at eval step) and the rest (structural modules/files) is commited to git (synchronised by `git checkout` at eval step).

However, we need to agree to run the training with the code to be in synch with git (which kinda makes sense)

Reply:
ok, addressed it in 48c9fe2